packages/familie-header/src/Søk.tsx

```diff
@@ -0,0 +1,46 @@
+import React, { useState } from 'react';
+import IkonSøk from './icons/IkonSøk';
+import './Søk.less';
+
+export interface SøkProps {
+    onSøk: (value: string) => void;
+    validator?: (value: string)=> boolean;
+    onGyldigVerdi?: (value: string)=> void;
+}
+
+const Søk = ({ onSøk, validator, onGyldigVerdi }: SøkProps) => {
+    const [value, setValue] = useState('');
+
+    const søk = () => {
+        if (value.length > 0) {
+            onSøk(value);
+        }
+    };
+
+    const onKeyPress = (event: React.KeyboardEvent) => {
+        if (event.key === 'Enter' && value.length > 0) {
+            søk();
+        }
+    };
+
+    const onChange = (event: React.ChangeEvent) => {
+        const changed= (event.target as HTMLInputElement).value
+        setValue(changed);
+        if(validator?.(changed)){
+            onGyldigVerdi?.(changed);
+        }
+    };
+
+    return (
+        <div>
+            <div className='søk_container'>
+                <input className='søkefelt' onChange={onChange} onKeyPress={onKeyPress} value={value} />
```

Comment:
I think we should use fnr-validator from nav to validate the input. So we can do all the input and validation internally in the component.

Reply:
because it is a shared search component which may need different validator in different project, so I think it is better to make it customizable.

Reply:
Fair enough! 👍